articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+
+### Preparing the input file
+In order to run this job you need to upload a list of users and passwords to check. This list must be provided as a CSV (coma separated value) where each row must has te following format:
+```
+{username},{email},{password}
+```
+
+#### Format considerations:
+- Must not has header row
+- Each row *must* have a maximum of 400 characters
+- If any of the values contain a coma, the whole value must be quoted
+- Email and username are optional, but you must provide one of them (and not both)
+
+The following JSON schema describes a row and might be used to validate it before uploading:
+```js
+   {
+     type: 'object',
+     anyOf: [
+       {
+         properties: {
+           email: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Email to check'
+           },
+           plain_text_password: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Password to check'
+           }
+         },
+         required: ['plain_text_password', 'email'],
+         additionalProperties: false
+       },
+       {
+         properties: {
+           username: {
+            type: 'string',
+            minLength: 1,
+            maxLength: 100,
+            description: 'Username to check'
+           },
+           plain_text_password: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Password to check'
+           }
+         },
+         required: ['plain_text_password', 'username'],
+         additionalProperties: false
+       }
+     ]
+   }
+```
+
+##### File Example
+```
+john.doe0@example.com,,secret
+,john.doe2,secret
+john.doe3@example.com,,"secret, really secret"
+,john.doe2,secr et
+```
+
+These rows are all valid input rows. Note the quotation around "secret, really secret"
+because that password contains a coma. Whitespace does not need quotation as stated on
+4th line.
+
+### Starting the job
+Once you have the file ready, you are ready to start a job to check the file against your database connection. To do so you need to execute a `POST` request against `/api/passwords-checking` with encoding type `multipart/form-data`.
+
+Your request should contain the following parameters:
+
+- passwords (the csv file, detailed above, that you are uploading, which contains data to check)
+- connection_id (a string, the connection id of the connection from where the users will be checked)
+
+If it works, you will get a response similar to the following one:
+
+```js
+{
+    "id":"job_abcdef1234567890",
+    "status":"pending",
+    "type":"passwords_checking",
+    "connection":"abcd"
+}
+```
+
+The returned entity represents the passwords checking job. At this point the job is queued on our system and will be started
```

Comment:
The returned entity represents the job in progress. At this point the job is queued. Once the job completes you will get a log entry `fpc` (for failure) or `spc` (for success). You can query the logs through [Management API2](/api/management/v2) or the [Dashboard](${manage_url}/#/logs). You can also query job status anytime using the appropriate endpoint as shown below.